ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -457,7 +523,12 @@ static bool ShouldHandleVnodeOpEvent(
         
         // TODO(Mac): why do we need to check rootIndex > 0 here?
         // If root was null, we would have already exited. And if not null, it can't be < 0.
-        if (rootIndex >= 0)
+        //
+        // TODO(Mac): Double check that rules are properly enforced when filedFlaggedAsInVirtualizationRoot
+        // is false.  These checks are currently skipped when filedFlaggedAsInVirtualizationRoot is false to
+        // allow providers to create new files inside of their root before starting the virtualization
```

Comment:
Why do we want to allow that?

Reply:
The problem we have right now is that `gvfs mount` will always attempt to copy in the hooks.

This code didn't cause a problem before because prior to my change above (to proceed even when `FileFlags_IsInVirtualizationRoot` is not set on the file), the kext would bail out before here because neither the .git folder nor the hooks file have the flag set.

I think that once we have the all of the FileOp\Vnode event handling wired up we'll need to revisit how to handle these offline scenarios.

I've filed a follow up issue (#182) for this